### PR TITLE
Fix job_template overwrites provider_type (#181)

### PR DESCRIPTION
**Pull request by m-bucher**
Apply default `provider_type` after template has been parsed

### CODE REVIEW

File: modules/foreman_job_template.py [hunk 1]
```diff
@@ -405,7 +409,9 @@ def main():
             module.fail_json(msg="Cannot use '*' as a job template name!")
         # module params are priorized
         parsed_dict.update(entity_dict)
-        entity_dict = parsed_dict
+        # make sure certain values are set
+        entity_dict = template_defaults.copy()
```
Comment (mdellweg) [inline modules/foreman_job_template.py]: Kodos for safe programming! Even if it's rather unlilkely that the template_defaults be reused.

**Comment by mdellweg** [comment]
Do you think, this is testable?

**Comment by m-bucher** [comment]
@mdellweg testing is tricky.
I tried adding a simple Ansible job_template once with explicit provider_type 'Ansible' and once without.
Without my above fix the test only fails when recording the tests, because then it would detect a `changed: true` for the second run because of the different provider_type.

However, in the normal case when running the test with recorded VCR-data, the test passes even without my fix, because the test does not check the request-bodies.
The request-body for adding with provider_type as module-param having `provider_type: Ansible` and the request-body without  module-param provider_type having `provider_type: SSH`.

**Comment by mdellweg** [comment]
The test should most definitely be matching requests depending on the body.

**Comment by mdellweg** [comment]
Ahh, #137 deactivated the body matcher. And i didn't  realize then.
@sean797 You removed it. We want it back. ;)

**Comment by sean797** [comment]
I think were have re-enable it in another PR, it's going to require some tests are re-recorded.

**Comment by m-bucher** [comment]
@sean797, @mdellweg I took the liberty of re-enabling it here, as well.
Getting all tests to run again took a little effort, though.
On the plus side, I found two python3 compatibility-bugs and was able to fix them

**Pull request merged**